site_config_client/__init__.py

```diff
@@ -22,6 +22,25 @@ def build_url(self, endpoint):
         full_path = urljoin(self.base_url, endpoint)
         return full_path
 
+    def create_site(self, domain_name, site_uuid=None):
+        """
+        Create a new site.
+        """
+        params = {'domain_name': domain_name}
+        if site_uuid:
+            params['uuid'] = site_uuid
+
+        auth_headers = {'Authorization': 'Token {}'.format(self.api_token)}
+        response = requests.post(self.build_url('v1/site/'),
```

Comment:
Network requests should never be made without specifying a timeout. By default `requests` will hang forever if the service it's trying to connect to isn't available.